jackrabbit-jcr2spi/src/main/java/org/apache/jackrabbit/jcr2spi/hierarchy/ChildNodeEntriesImpl.java

```diff
@@ -704,15 +701,10 @@ public LinkedEntries.LinkNode next() {
                 return n;
             }
 
+            @Override
             public void remove() {
                 throw new UnsupportedOperationException("remove");
             }
-
```

Comment:
I stumbled across this as well; the original code apparently tried to protect itself against modifcations through other methods. Can we really eliminate this?

Reply:
I've added test for concurrent modification. Was passing when I accessed the modCount field via reflection but doesn't work when I overlay the field. Must be a stupid mistake but I can't see it 🤔